Manage config file

exploide opened the pull request:
Since it was really easy to implement, I fixed #23 w/o waiting for your feedback. Feel free to decline or request changes ;-)

This introduces a role variable that allows to set global options for certbot that go into `/etc/letsencrypt/cli.ini`.

exploide commented:
Ok, I don't see why the Ubuntu tests fail. On the first glance, this looks unrelated.

oxyc commented:
Yeah it's an upstream bug https://github.com/ansible/ansible/issues/23530. Also https://github.com/geerlingguy/drupal-vm/issues/1302.

Unfortunately I don't think the more ideal fix of `{{ ansible_user|default(ansible_user_id) }}` will work on Ubuntu 12.04 but maybe worth a try and see what travis says. If the variable defaulted to `ansible_user_id`, the docs might need to highlight even more that the user should be changed to a non-root user account, as that's the default otherwise.

exploide commented:
Ah, thanks for the explanation and pointing this out. Then it is in fact unrelated and further action concerning this is out of scope of this PR.

exploide commented:
Is there anything I can do to help getting this merged?

oxyc commented:
I believe the fix is https://github.com/geerlingguy/ansible-role-backup/commit/72b994510ba67b04dc6235eead7f912f1e36c489

exploide commented:
Ok, I incorporated this change. Unfortunately, it seems CentOS 6 still can't find the variable.

tterranigma commented:
I think this will work:
```yaml
certbot_auto_renew_user: "{{ lookup('env', 'USER') }}"
```

exploide commented:
Finally :D thanks @tterranigma 

Hopefully it's ready to merge now @geerlingguy

exploide commented:
I rebased and cleaned up this PR. While this has nothing directly to do with using the staging server, it generally allows deploying the `/etc/letsencrypt/cli.ini` configuration file with whatever options one wants.

@geerlingguy do you want this to go in? If yes feel free to merge, but also feel free to close otherwise. Meanwhile I'm using an own role tailored to the needs of my infrastructure, so this is no longer important to me but I wanted to keep the contribution usable and I'm interested in resolving this PR.

kzap commented:
could this be merged @geerlingguy ?
